packages/clerk-js/src/ui/components/OrganizationProfile/InviteMembersForm.tsx

```diff
@@ -89,24 +89,34 @@ export const InviteMembersForm = (props: InviteMembersFormProps) => {
       .inviteMembers({ emailAddresses: emailAddressField.value.split(','), role: roleField.value as MembershipRole })
       .then(onSuccess)
       .catch(err => {
+        removeInvalidEmails(err);
+
         if (isClerkAPIResponseError(err) && err.errors?.[0]?.code === 'duplicate_record') {
           const unlocalizedEmailsList = err.errors[0].meta?.emailAddresses || [];
 
           // Create a localized list of email addresses
           const localizedList = createListFormat(unlocalizedEmailsList, locale);
           setLocalizedEmails(localizedList);
-
-          // Remove any invalid email address
-          const invalids = new Set(unlocalizedEmailsList);
-          const emails = emailAddressField.value.split(',');
-          emailAddressField.setValue(emails.filter(e => !invalids.has(e)).join(','));
         } else {
           setLocalizedEmails(null);
           handleError(err, [], card.setError);
         }
       });
   };
 
+  const removeInvalidEmails = (err: any) => {
```

Comment:
Can we do this and simple instead of calling this function to do 
```suggestion
  const removeInvalidEmails = (err: ClerkAPIError) => {
```

```tsx

if (isClerkAPIResponseError(err)) {
  removeInvalidEmails(err);
}
```

Reply:
Updated!